extension/test/integration/get-origin-keys.handler.spec.js

```diff
@@ -0,0 +1,119 @@
+    'then should set custom field getOriginKeysResponse ' +
+    'and interface interaction with type getOriginKeysRequest', async () => {
+    const getOriginKeysRequestDraft = {
+      originDomains: [
+        "http://localhost:8080",
+        "http://localhost:8081"
+      ]
+    }
+    const paymentDraft = {
+      amountPlanned: {
+        currencyCode: 'EUR',
+        centAmount: 1000,
+      },
+      paymentMethodInfo: {
+        paymentInterface: c.CTP_ADYEN_INTEGRATION
+      },
+      custom: {
+        type: {
+          typeId: "type",
+          key: c.CTP_WEB_COMPONENTS_PAYMENT_TYPE_KEY
+        },
+        fields: {
+          getOriginKeysRequest: JSON.stringify(getOriginKeysRequestDraft)
+        }
+      }
+    }
+
+    const { statusCode, body: payment } = await ctpClient.create(ctpClient.builder.payments, paymentDraft)
+    expect(statusCode).to.equal(201)
+
+    const { getOriginKeysRequest, getOriginKeysResponse } = payment.custom.fields
+    expect(getOriginKeysRequest).to.be.deep.equal(JSON.stringify(getOriginKeysRequestDraft))
+
+    const interfaceInteraction = payment.interfaceInteractions
+      .find(interaction => interaction.fields.type === c.CTP_INTERACTION_TYPE_GET_ORIGIN_KEYS)
+    expect(interfaceInteraction).to.not.undefined
+    expect(getOriginKeysResponse).to.be.deep.equal(interfaceInteraction.fields.response)
+
+    expect(JSON.parse(interfaceInteraction.fields.request)).to.be.deep.equal({
+      merchantAccount: process.env.ADYEN_MERCHANT_ACCOUNT,
+      ...getOriginKeysRequestDraft
+    })
+
+    const interfaceInteractionResponse = JSON.parse(interfaceInteraction.fields.response)
+    expect(interfaceInteractionResponse.originKeys).to.exist
+  })
+
+  it('given a payment ' +
+    'when getOriginKeysRequest is not includes originDomains array and custom field getOriginKeysResponse is not set ' +
+    'then should set custom field with a serialized error getOriginKeysResponse with ' +
+    'and interface interaction with type getOriginKeysRequest', async () => {
+    const getOriginKeysRequestDraft = { } //an invalid request
+
+    const paymentDraft = {
+      amountPlanned: {
+        currencyCode: 'EUR',
+        centAmount: 1000,
+      },
+      paymentMethodInfo: {
+        paymentInterface: c.CTP_ADYEN_INTEGRATION
+      },
+      custom: {
+        type: {
+          typeId: "type",
+          key: c.CTP_WEB_COMPONENTS_PAYMENT_TYPE_KEY
+        },
+        fields: {
+          getOriginKeysRequest: JSON.stringify(getOriginKeysRequestDraft)
+        }
+      }
+    }
+
+    const { statusCode, body: payment } = await ctpClient.create(ctpClient.builder.payments, paymentDraft)
+    expect(statusCode).to.equal(201)
+
+    const { getOriginKeysRequest, getOriginKeysResponse } = payment.custom.fields
+    expect(getOriginKeysRequest).to.be.deep.equal(JSON.stringify(getOriginKeysRequestDraft))
+    expect(JSON.parse(getOriginKeysResponse)).to.be.deep.equal({
```

Comment:
I don't think we care about the response in case of error. We care that the key is not set and sth is stored in here. This error can be changed any time and then our test will fail.

Reply:
I removed those tests, This is the same change, #232 (comment)